app/src/PostDetail/index.js

```diff
@@ -0,0 +1,115 @@
+import * as React from "react";
+
+import ReactMarkdown from "react-markdown";
+import { useParams } from "react-router-dom";
+
+import * as apiClient from "../apiClient";
+
+import styles from "./styles.module.scss";
+
+const PostDetail = () => {
+  const [post, setPost] = React.useState();
+  const [comments, setComments] = React.useState();
+  const { id } = useParams();
+
+  const getPost = React.useCallback(
+    () => apiClient.getPost(id).then(setPost),
+    [id],
+  );
+
+  const getComments = React.useCallback(
+    () => apiClient.getComments(id).then(setComments),
+    [id],
+  );
+
+  const addComment = (comment) =>
+    apiClient.addComment(comment).then(getComments);
+
+  React.useEffect(() => {
+    id !== undefined && getPost() && getComments();
+  }, [id, getPost, getComments]);
+
+  const onSubmit = (event) => {
+    const form = event.currentTarget;
+    const {
+      content: { value: content },
+      author: { value: author },
+    } = form.elements;
+
+    event.preventDefault();
+    addComment({ id, content, author });
+    form.reset();
+  };
+
+  const hrLine = (
```

Comment:
[suggest] I would move the styling to your CSS module, and just use `<hr className={styles.hr}>` directly in the JSX. I'm personally not a fan of this way of injecting JSX.